packages/grid/_modules_/grid/components/column-header-item.tsx

```diff
@@ -55,42 +67,57 @@ export const ColumnHeaderItem = React.memo(
 
     return (
       <div
-        className={cssClass}
+        className={classnames(
+          HEADER_CELL_CSS_CLASS,
+          showColumnRightBorder ? 'MuiDataGrid-withBorder' : '',
+          column.headerClassName,
+          column.headerAlign === 'center' && 'MuiDataGrid-colCellCenter',
+          column.headerAlign === 'right' && 'MuiDataGrid-colCellRight',
+          { 'MuiDataGrid-colCellSortable': column.sortable },
+        )}
         key={column.field}
         data-field={column.field}
         style={{
           width,
           minWidth: width,
           maxWidth: width,
-          maxHeight: headerHeight,
-          minHeight: headerHeight,
         }}
         role="columnheader"
         tabIndex={-1}
         aria-colindex={colIndex + 1}
         {...ariaSort}
       >
-        {column.type === 'number' && (
-          <ColumnHeaderSortIcon
-            direction={column.sortDirection}
-            index={column.sortIndex}
-            hide={column.hideSortIcons}
-          />
-        )}
-        {headerComponent || (
-          <ColumnHeaderTitle
-            label={column.headerName || column.field}
-            description={column.description}
-            columnWidth={width}
-          />
-        )}
-        {column.type !== 'number' && (
-          <ColumnHeaderSortIcon
-            direction={column.sortDirection}
-            index={column.sortIndex}
-            hide={column.hideSortIcons}
-          />
-        )}
+        <div
+          className="MuiDataGrid-colCell-draggable"
+          draggable={
+            !disableColumnReorder && !!handleDragStart && !!handleDragEnter && !!handleDragOver
+          }
+          onDragStart={handleDragStart}
```

Comment:
either do the check here or spread an object with those functions including draggable
code is twice the size

Reply:
Done, I've added all of them in a `dragConfig` object and spread it.